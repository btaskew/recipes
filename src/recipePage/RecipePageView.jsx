import React from 'react';

import {Alert} from 'react-bootstrap';
import Page from 'components/Page';
import InputForm from 'components/InputForm';
import IngredientsList from 'recipePage/IngredientsList';

function RecipePageView(props) {
    return (
        <Page heading={props.heading} loading={props.loading}>
            {props.ingredients && (
                <React.Fragment>
                    <InputForm
                        submitForm={props.addIngredient}
                        buttonText="Add"
                        inputText="Enter ingredient name"
                    />
                    <hr />
                    <IngredientsList ingredients={props.ingredients} />
                </React.Fragment>
            )}

            {props.error && <Alert bsStyle="danger">{props.errorMessage}</Alert>}
        </Page>
    );
}

export default RecipePageView;
