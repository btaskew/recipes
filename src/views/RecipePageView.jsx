import React from 'react';

import Page from 'components/utils/Page';
import IngredientsList from 'components/IngredientsList';
import InputForm from 'components/utils/InputForm';

function RecipePageView(props) {
    return (
        <Page
            heading={props.heading}
            loading={props.loading}
            error={props.error}
            errorMessage={props.errorMessage}
            noResult={props.noResult}
        >
            {props.ingredients && (
                <React.Fragment>
                    <InputForm
                        submitForm={props.addIngredient}
                        buttonText="Add"
                        inputText="Enter ingredient name"
                    />
                    <IngredientsList ingredients={props.ingredients} />
                </React.Fragment>
            )}
        </Page>
    );
}

export default RecipePageView;
