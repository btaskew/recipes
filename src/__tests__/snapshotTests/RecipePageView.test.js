import React from 'react';
import renderer from 'react-test-renderer';

import RecipePageView from 'recipePage/RecipePageView';
import {MemoryRouter} from 'react-router';

describe('Recipe page', () => {
    it('renders loading correctly', () => {
        const recipePage = renderer
            .create(
                <MemoryRouter>
                    <RecipePageView loading={true} heading="Loading" />
                </MemoryRouter>
            )
            .toJSON();
        expect(recipePage).toMatchSnapshot();
    });

    it('renders an error correctly', () => {
        const recipePage = renderer
            .create(
                <MemoryRouter>
                    <RecipePageView
                        loading={true}
                        heading="Error"
                        error={true}
                        errorMessage="Error"
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(recipePage).toMatchSnapshot();
    });

    it('renders with ingredients correctly', () => {
        const recipePage = renderer
            .create(
                <MemoryRouter>
                    <RecipePageView
                        loading={true}
                        heading="Error"
                        ingredients={[]}
                        addIngredient={() => {}}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(recipePage).toMatchSnapshot();
    });
});
