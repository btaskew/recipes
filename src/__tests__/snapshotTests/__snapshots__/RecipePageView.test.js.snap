// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe page renders an error correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Error
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;

exports[`Recipe page renders loading correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Loading
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;

exports[`Recipe page renders with ingredients correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Error
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;
